js/bundle_test.go

```diff
@@ -114,12 +114,18 @@ func TestNewBundle(t *testing.T) {
 		}
 	})
 	t.Run("CompatibilityMode", func(t *testing.T) {
+		compatibilityTestSrc := `
+		module.exports.default = function() {};
+		if ([[1,2],[3,4]].flatten()[4]  == 4) {
+			throw Error("Array.flatten doesn't work")
```

Comment:
This would never fail, no? It would always be undefined, even if `flatten()` works, unless you meant `[3] === 4`.

But this would be better checked with `if (typeof Array.prototype.flatten !== 'function') ...`.

Also maybe we should use a different object to test this, since [`flatten` was renamed to `flat` in corejs v3](https://github.com/zloirock/core-js/issues/405), and it would avoid us having to update this when we update to v3.

Reply:
> But this would be better checked with `if (typeof Array.prototype.flatten !== 'function') ....`

We should have both of these. First check if it's actually a function, and fast fail if it's no, then whether the behavior is at least somewhat along the lines of what one could expect.

> Also maybe we should use a different object to test this, since flatten was renamed to flat in corejs v3, and it would avoid us having to update this when we update to v3.

`Array.prototype.flat` is the actual name according to the specification. We should probably use that name as well.